### ISSUE TITLE
Error when running "imagetest" example

### ISSUE THREAD

**Issue by bvssvni** [PistonDevelopers/gfx_graphics#135]
```
RUST_BACKTRACE=1 cargo run --example imagetest
     Running `target/examples/imagetest`
thread '<main>' panicked at 'assertion failed: data.len() * esize + offset_bytes <= buf.get_info().size', /Users/sven/.cargo/git/checkouts/gfx-rs-93bce98024942039/master/src/render/lib.rs:255
stack backtrace:
   1:        0x10e916a77 - sys::backtrace::write::h159c9303aa56d993QnB
   2:        0x10e942a33 - failure::on_fail::h30a5d63a5824ccdbmFK
   3:        0x10e87ada8 - rt::unwind::begin_unwind_inner::h83398b7af9be1f1fCmK
   4:        0x10af498d7 - rt::unwind::begin_unwind::h12418617414705309073
   5:        0x10af69020 - render::Renderer<D>::update_buffer_vec::h11957271402520406697
   6:        0x10af68d3b - g2d::GraphicsBackEnd<'a, C>.BackEnd::tri_list::closure.7606
   7:        0x10af70662 - extern $u22$rust.call$u22$$u20$fn$LP$$RF$mut$u20$for$LT$$u27$r$GT$$u20$core..ops..FnMut$LP$$RF$$u27$r$u20$$u5b$f32$u5d$$RP$$u20$$u2b$$u20$$u27$static$C$$u20$$LP$$RF$$u27$static$u20$$u5b$f32$u5d$$C$$RP$$RP$::object_shim.7755::hd0e0a155049dfd3b
   8:        0x10af68bf2 - rectangle::Rectangle::draw::closure.7604
   9:        0x10af66d27 - g2d::GraphicsBackEnd<'a, C>.BackEnd::tri_list::h10387419007337182272
  10:        0x10af66904 - rectangle::Rectangle::draw::h11168062705347684041
  11:        0x10af6006d - main::closure.7507
  12:        0x10af5fc38 - g2d::G2D::draw::h12446179003519822330
  13:        0x10af1d526 - main::h851c6c8299080913uaa
  14:        0x10e9ba1a9 - rust_try_inner
  15:        0x10e9ba196 - rust_try
  16:        0x10e94347d - rt::lang_start::h06967cab2a70aa6bnzK
  17:        0x10af1d65f - main
An unknown error occurred
```

**Comment by bvssvni** [comment]
Printing out the size using `.get_info().size`:

```
TEST buffer_pos: 8192, vertices: 12
TEST buffer_pos: 0, vertices: 12
```

**Comment by bvssvni** [comment]
It is a bit curious that the size of the buffer becomes `0`.

**Comment by bvssvni** [comment]
Printing this out before and after `update_buffer_vec` reveals that it changes during the update:

```
TEST 1 buffer_pos: 8192, vertices: 12
TEST 2 buffer_pos: 0, vertices: 12
TEST 1 buffer_pos: 0, vertices: 12
```

**Comment by bvssvni** [comment]
Overriding Gfx, and it prints the same size at the end of `update_buffer_vec`, but `0` after the function call.

```
TEST 1 buffer_pos: 8192, vertices: 12
TEST 1 buf.get_info().size 8192
TEST 2 buf.get_info().size 8192
TEST 2 buffer_pos: 0, vertices: 12
TEST 1 buffer_pos: 0, vertices: 12
```

**Comment by bvssvni** [comment]
My guess is this got something to do with the deref `*buffer_pos` when calling the function.

**Comment by kvark** [comment]
Supposedly fixed by #136
